llvm/include/llvm/Analysis/DXILResource.h

```diff
@@ -574,6 +577,80 @@ class DXILResourceBindingWrapperPass : public ModulePass {
 
 ModulePass *createDXILResourceBindingWrapperPassPass();
 
+enum class ResourceCounterDirection {
+  Increment,
+  Decrement,
+  Unknown,
+  Invalid,
+};
+
+class DXILResourceCounterDirectionMap {
+  std::vector<std::pair<dxil::ResourceBindingInfo, ResourceCounterDirection>>
+      CounterDirections;
+
+public:
+  void populate(Module &M, DXILBindingMap &DBM);
+
+  ResourceCounterDirection
+  operator[](const dxil::ResourceBindingInfo &Info) const {
+    auto Lower = llvm::lower_bound(
+        CounterDirections, Info,
+        [](const auto &LHS, const auto &RHS) { return LHS.first < RHS; });
+
+    if (Lower == CounterDirections.end()) {
+      return ResourceCounterDirection::Unknown;
+    }
+
+    if (Lower->first != Info) {
+      return ResourceCounterDirection::Unknown;
+    }
```

Comment:
https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements

```suggestion
    if (Lower == CounterDirections.end())
      return ResourceCounterDirection::Unknown;
    if (Lower->first != Info)
      return ResourceCounterDirection::Unknown;
```

Reply:
lol I swear one day I'll get this bashed into my head. I notice it on reviews but when I'm writing code the muscle memory takes over